edx_lint/pylint/unicode_check.py

```diff
@@ -84,4 +97,12 @@ def process_tokens(self, tokens):
                     # Valid JSON isn't human-readable, leave it alone.
                     continue    # pragma: no cover   grr.. https://github.com/nedbat/coveragepy/issues/198
 
+                if re.match(r"^(def|class) [a-zA-Z0-9_]*\(.*", previous_line):
```

Comment:
although this may not always work, it may be good enough and may be simpler than other more correct solutions that would involve a lot more refactoring

Reply:
A small tweak to the regex:
```
re.match(r"^\s*(def|class) [a-zA-Z0-9_]+\(", previous_line)
```

Reply:
Also, let's move this re.match check before the json.loads attempt.  No point trying to json-parse all the docstrings in the code.

Reply:
good point, i'll move it before.  
I think it may have worked without the s due to indents being another token but i'll add it just in case. we still want the .* at the end though since this is match not search i think

Reply:
`match` only needs to match the start of the string (and in fact, `^` is redundant for match), so the trailing `.*` can be removed.

Reply:
ah ok good to know, thanks